debian/patches/backport-ubuntu-pro-rename.patch

```diff
@@ -0,0 +1,2522 @@
++            ", ".join(enable_beta),
++        )
++    configure_pro(token=token, enable=pro_section.get("enable"))
++
++
++def _auto_attach(pro_section: dict):
++
++    # pylint: disable=import-error
++    from uaclient.api.exceptions import AlreadyAttachedError, UserFacingError
++    from uaclient.api.u.pro.attach.auto.full_auto_attach.v1 import (
++        FullAutoAttachOptions,
++        full_auto_attach,
++    )
++
++    # pylint: enable=import-error
++
++    enable = pro_section.get("enable")
++    enable_beta = pro_section.get("enable_beta")
++    options = FullAutoAttachOptions(
++        enable=enable,
++        enable_beta=enable_beta,
++    )
++    try:
++        util.log_time(
++            logfunc=LOG.debug,
++            msg="Attaching to Ubuntu Pro",
++            func=full_auto_attach,
++            kwargs={"options": options},
++        )
++    except AlreadyAttachedError:
++        if enable_beta is not None or enable is not None:
++            # Only warn if the user defined some service to enable/disable.
++            LOG.warning(
++                "The instance is already attached to Pro. Leaving enabled"
++                " services untouched. Ignoring config directives"
++                " ubuntu_pro: enable and enable_beta"
++            )
++    except UserFacingError as ex:
++        msg = "Error during `full_auto_attach`: %s" % ex.msg
++        LOG.error(msg)
++        raise RuntimeError(msg) from ex
++
++
++def handle(
++    name: str, cfg: dict, cloud: Cloud, log: logging.Logger, args: list
++) -> None:
++    pro_section = None
++    deprecated = list(DEPRECATED_KEYS.intersection(cfg))
++    if deprecated:
++        if len(deprecated) > 1:
++            raise RuntimeError(
++                "Unable to configure Ubuntu Pro. Multiple deprecated config"
++                " keys provided: %s" % ", ".join(deprecated)
++            )
++        LOG.debug(
++            "Deprecated configuration key(s) provided: %s."
++            ' Expected "ubuntu_pro"; will attempt to continue.',
++            ", ".join(deprecated),
++        )
++        pro_section = cfg[deprecated[0]]
++    if "ubuntu_pro" in cfg:
++        # Prefer ubuntu_pro over any deprecated keys when both exist
++        if deprecated:
++            LOG.debug(
++                "Ignoring deprecated key %s and preferring ubuntu_pro config",
++                deprecated[0],
++            )
++        pro_section = cfg["ubuntu_pro"]
++    if pro_section is None:
++        LOG.debug(
++            "Skipping module named %s, no 'ubuntu_pro' configuration found",
++            name,
++        )
++        return
++    elif not isinstance(pro_section, dict):
++        msg = (
++            "'ubuntu_pro' should be a dict, not a"
++            " %s" % type(pro_section).__name__
++        )
++        LOG.error(msg)
++        raise RuntimeError(msg)
++    validate_cloudconfig_schema(cfg, schema)
++    if "commands" in pro_section:
```

Comment:
Was this behavior already broken in Xenial? If not, why are we breaking it here?

Reply:
Great point. pulling this element in wholesales doesn't make sense to introduce this error. We should continue to support "commands" in xenial.